subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheExternalProcessInstrumentationIntegrationTest.groovy

```diff
@@ -0,0 +1,564 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache
+
+import org.gradle.process.ShellScript
+
+class ConfigurationCacheExternalProcessInstrumentationIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
+    // Note that all tests use a relative path to the script because its absolute path may contain
+    // spaces and it breaks logic String.execute which splits the given string at spaces without
+    // any options to escape the space.
+    ShellScript baseScript = ShellScript.builder().printEnvironmentVariable('FOOBAR').printWorkingDir().writeTo(testDirectory, "test")
+
+    def setup() {
+        testDirectory.createDir(pwd)
+    }
+
+    def "#title is intercepted in groovy build script"(VarInitializer varInitializer) {
```

Comment:
🟡 I'm missing some test proving that regular `execute` methods (unrelated to process execution) are passing through and no problems are reported.

Reply:
> I'm missing some test proving that regular execute methods

Indeed. I've added a few more tests for that

Reply:
Great!